### PR TITLE
Consolidate the rest of se23 scss styles

### PR DESCRIPTION
**Pull request by sophschneider**
### WHAT is this pull request doing?

* Consolidates `se23` styles in scss files that were either missed or added back in as part of bad rebases
* Renames stylelint comments that include `se23`

### How to 🎩

Tophatting links are in the file changes per component

### CODE REVIEW

File: polaris-react/src/components/ActionList/ActionList.scss [hunk 1]
```diff
@@ -96,12 +96,9 @@ $indented-item-width: calc(100% - #{$indented-item-margin});
   &.disabled {
     background-image: none;
     color: var(--p-color-text-disabled);
+    cursor: default;
 
-    #{$se23} & {
-      cursor: default;
-    }
-
-    #{$se23} &:hover {
```
Comment (sophschneider) [inline polaris-react/src/components/ActionList/ActionList.scss:104]: * [This PR](https://5d559397bae39100201eedc1-tehbqainqm.chromatic.com/?path=/story/all-components-actionlist--with-an-icon-and-a-suffix)
* [Production](https://storybook.polaris.shopify.com/?path=/story/all-components-actionlist--with-an-icon-and-a-suffix&globals=polarisSummerEditions2023:true)

Screenshot 2023-08-25 at 1 45 50 PM

File: polaris-react/src/components/Checkbox/Checkbox.scss [hunk 2]
```diff
@@ -40,14 +40,11 @@
...
-          var(--p-color-border-interactive-focus);
-        outline-offset: var(--p-space-025);
-        background-color: var(--p-color-bg-input-hover-experimental);
-        border-width: 0;
-      }
+      outline: var(--p-border-width-2) solid
+        var(--p-color-border-interactive-focus);
+      outline-offset: var(--p-space-025);
+      background-color: var(--p-color-bg-input-hover-experimental);
+      border-width: 0;
```
Comment (sophschneider) [inline polaris-react/src/components/Checkbox/Checkbox.scss:47]: * [This PR](https://5d559397bae39100201eedc1-tehbqainqm.chromatic.com/?path=/story/all-components-checkbox--default)
* [Production](https://storybook.polaris.shopify.com/?path=/story/all-components-checkbox--default&globals=polarisSummerEditions2023:true)

Screenshot 2023-08-25 at 1 48 56 PM

File: polaris-react/src/components/Select/Select.scss [hunk 3]
```diff
@@ -43,13 +43,6 @@
     fill: var(--p-color-icon-disabled);
   }
 
-  #{$se23} & {
-    // stylelint-disable-next-line -- se23 specificity
-    .Icon svg {
-      fill: var(--p-color-icon-disabled);
-    }
-  }
```
Comment (sophschneider) [inline polaris-react/src/components/Select/Select.scss:51]: [This PR](https://5d559397bae39100201eedc1-tehbqainqm.chromatic.com/?path=/story/all-components-select--disabled)
[Production](https://storybook.polaris.shopify.com/?path=/story/all-components-select--disabled&globals=polarisSummerEditions2023:true)

Screenshot 2023-08-25 at 1 51 20 PM

**Review by laurkim — APPROVED** [review]
🙌 🚀

**Pull request merged**